### PR TITLE
Init compilers

### PR DESCRIPTION
**Pull request by kuzdogan**
Adds the optional init functin to VerificationService to download all compilers ahead. This is useful for the migration server because when we start hitting the server with a lot of verification requests, they translate into solc download requests and the solc server gets overloaded (like 20 req per chain x 100s chains). 

## Summary by CodeRabbit

- **New Features**
  - Added exponential backoff retry logic for fetching resources to improve reliability.
  - Introduced a new configuration option to control the initialization of compilers.

- **Improvements**
  - Enhanced error handling and logging for fetching resources.
  - Improved clarity and efficiency in the initialization of storage services.
  - Added detailed logging for database pool setup and health checks.

- **Bug Fixes**
  - Removed redundant configuration settings to streamline test environments.

- **Refactor**
  - Updated multiple functions to incorporate exponential backoff strategy.
  - Refined the `VerificationService` to use new options for better configurability.

### CODE REVIEW

**Comment by kuzdogan** [comment]
@coderabbitai please review

File: services/server/src/server/services/VerificationService.ts [hunk 1]
```diff
@@ -7,25 +7,75 @@ import {
...
+
+      const chunkSize = 10; // Download in chunks to not overload the Solidity server all at once
+      for (let i = 0; i < solcList.length; i += chunkSize) {
+        const chunk = solcList.slice(i, i + chunkSize);
+        const promises = chunk.map((solcVer) => {
+          const now = Date.now();
+          return downLoadFunc(solcVer).then(() => {
+            logger.debug(
+              `Downloaded compiler ${solcVer} in ${Date.now() - now}ms`,
+            );
```
Comment (marcocastignoli) [inline services/server/src/server/services/VerificationService.ts:66]: It's not clear that there is a caching mechanism inside getSolcJs/getSolc functions such that if the compiler already exists, it is not downloaded again. I would add a comment or change the name of the function `downloadIdMissing` and log "downloaded" if it was downloaded OR "Already exists" OR something generic

**Review by marcocastignoli — APPROVED** [review]
Looks good! 🚀 Only one small comment, but it's not important.

File: services/server/src/server/services/VerificationService.ts (hunk 1 above)
Comment (kuzdogan) [reply services/server/src/server/services/VerificationService.ts]: If it's already found it's logged inside the getSolc/getSolcjs functions. Still I added `Downloaded (or found existing) compiler` It should be also apparent in how long it took to find/download it.

**Pull request merged**